src/tests/ftest/util/cmocka_utils.py

```diff
@@ -1,18 +1,58 @@
 """
-  (C) Copyright 2022-2023 Intel Corporation.
+  (C) Copyright 2022-2024 Intel Corporation.
 
   SPDX-License-Identifier: BSD-2-Clause-Patent
 """
 import os
 
 from agent_utils import include_local_host
 from command_utils import ExecutableCommand
-from command_utils_base import EnvironmentVariables
+from command_utils_base import BasicParameter, EnvironmentVariables
 from exception_utils import CommandFailure
 from results_utils import Job, Results, TestName, TestResult, create_xml
 from run_utils import get_clush_command, run_local, run_remote
 
 
+def get_cmocka_command(command, parameters=None):
+    """Get an ExecutableCommand representing the provided command string.
+
+    Adds detection of any bad keywords in the command output that, if found, will result in a
+    command failure.
+
+    Args:
+        command (str): the command path to use to create the CmockaCommand
+        parameters (str): parameters to use to create the CmockaCommand
+
+    Returns:
+        ExecutableCommand: the object setup to run the command
+    """
+    keywords = ["Process received signal", "stack smashing detected", "End of error message",
+                "buffer overflow detected"]
+    path, executable = os.path.split(command)
+    command = CmockaCommand(path, executable, keywords)
+    command.parameters.value = parameters
+    return command
+
+
+class CmockaCommand(ExecutableCommand):
+    """Defines a object representing a cmocka test command."""
+
+    def __init__(self, path, executable, keywords):
+        """Create a CmockaCommand object.
+
+        Args:
+            path (str): the command path. This is excluded from the search when the process is
+                killed.
+            executable (str): the command executable. Also the string used to search for the process
+                when it is killed.
+            keywords (list): list of words used to mark the command as failed if any are found in
```

Comment:
Should we call `keywords` `check_results` to be consistent?

Reply:
Yeah, it's not a bad idea.